eda/src/autogluon/eda/analysis/shift.py

```diff
@@ -0,0 +1,347 @@
+
+    Parameters
+    ----------
+    classifier_class : an AutoGluon predictor, such as autogluon.tabular.TabularPredictor
+        The predictor (classifier) class to classify the source from target dataset, predictor class needs to support
+        binary classification
+    sample_label : str, default = 'xshift_label'
+        The label that will be used to indicate if the sample is from training or test
+    eval_metric : callable, default = autogluon.core.metrics.balanced_accuracy
+        Binary classification metric to use for the classifier 2 sample test
+    split : float, default = 0.5
+        Training/test split proportion for classifier 2 sample test
+    classifier_kwargs : dict, default = {}
+        The kwargs passed to the classifier, a member of classifier_class
+    """
+    def __init__(self,
+                 classifier_class,
+                 sample_label='xshift_label',
+                 eval_metric=balanced_accuracy,
+                 split=0.5,
+                 compute_fi = True,
+                 classifier_kwargs = {}
+                 ):
+        classifier_kwargs.update({'label': sample_label, 'eval_metric': eval_metric})
+        self.classifier = classifier_class(**classifier_kwargs)
+        self.classifier_class = classifier_class
+        self.split = split
+        self.sample_label = sample_label
+        self.eval_metric = eval_metric
+        self._is_fit = False
+        self._test = None
+        self.test_stat = None
+        self.has_fi = None
+        self.compute_fi = compute_fi
+
+    @staticmethod
+    def _make_source_target_label(data, sample_label):
+        """Turn a source, target pair into a single dataframe with label column"""
+        source, target = data[0].copy(), data[1].copy()
+        source.loc[:,sample_label] = 0
+        target.loc[:,sample_label] = 1
+        data = pd.concat((source, target))
+        return data
+
+    def fit(self, data, **kwargs):
+        """Fit the classifier for predicting if source or target and compute the 2-sample test statistic.
+
+        Parameters
+        ----------
+        data : pd.DataFrame, or tuple
+            either
+            - a dataframe with a label column where 1 = target and 0 = source
+            - a tuple of source dataframe and target dataframe
+        """
+        if isinstance(data, pd.DataFrame):
+            sample_label = self.sample_label
+            assert sample_label in data.columns, "sample_label needs to be a column of data"
+            assert self.split, "sample_label requires the split parameter"
+            data = data.copy() # makes a copy
+        else:
+            assert len(data) == 2, "Data needs to be tuple/list of (source, target) if sample_label is None"
+            data = self._make_source_target_label(data, self.sample_label) # makes a copy
+        if data.index.has_duplicates:
+            data.index = pd.RangeIndex(data.shape[0])
+        train = data.sample(frac=self.split)
```

Comment:
Let's fix the seed (`random_state=0`) so it is reproducible; otherwise it might be confusing to the users to get different results on repeated runs.
Open question: does it make sense to do stratification by label?

Reply:
It could make sense to stratify by label, that would ensure that there is enough of both source and target.  Especially helpful when there are few target samples (a reasonable scenario).